fbw-a32nx/src/systems/fmgc/src/flightplanning/MsfsFlightPlanSync.ts

```diff
@@ -0,0 +1,500 @@
+    [ApproachType.Unknown]: '',
+    [ApproachType.LocBackcourse]: 'BLOC',
+    [ApproachType.VorDme]: 'VDM',
+    [ApproachType.Fms]: '',
+    [ApproachType.Igs]: '',
+    [ApproachType.Ils]: 'ILS',
+    [ApproachType.Gls]: ' ',
+    [ApproachType.Loc]: 'LOC',
+    [ApproachType.Mls]: '',
+    [ApproachType.Ndb]: 'NDB',
+    [ApproachType.Gps]: 'GPS',
+    [ApproachType.NdbDme]: 'NDBDME',
+    [ApproachType.Rnav]: 'RNAV',
+    [ApproachType.Vortac]: '',
+    [ApproachType.Tacan]: '',
+    [ApproachType.Sdf]: '',
+    [ApproachType.Vor]: 'VOR',
+    [ApproachType.MlsTypeA]: '',
+    [ApproachType.Lda]: 'LDA',
+    [ApproachType.MlsTypeBC]: '',
+  };
+
+  private readonly listener: ViewListener.ViewListener;
+
+  private readonly rpcClient: FlightPlanRpcClient<A320FlightPlanPerformanceData>;
+
+  private readonly dataManager: DataManager;
+
+  private readonly listenerReady = Subject.create(false);
+
+  private readonly fmsRpcClientReady = Subject.create(false);
+
+  private readonly isReady = MappedSubject.create(([a, b]) => a && b, this.listenerReady, this.fmsRpcClientReady);
+
+  constructor(private readonly bus: EventBus) {
+    if (!isMsfs2024()) {
+      throw new Error('[MsfsFlightPlanSync] Cannot be instantiated in MSFS 2020');
+    }
+
+    console.log('[MsfsFlightPlanSync] Created');
+
+    this.listener = RegisterViewListener('JS_LISTENER_PLANNEDROUTE', () => {
+      console.log('[MsfsFlightPlanSync] Listener initialized');
+
+      this.listenerReady.set(true);
+    });
+
+    this.rpcClient = new FlightPlanRpcClient(bus, new A320FlightPlanPerformanceData());
+    this.rpcClient.onAvailable.on(() => this.fmsRpcClientReady.set(true));
+
+    this.dataManager = new DataManager(this.bus, this.rpcClient);
+
+    Wait.awaitSubscribable(this.isReady).then(() => {
+      console.log('[MsfsFlightPlanSync] Ready');
+
+      const autoLoadRoute = NXDataStore.get('CONFIG_AUTO_SIM_ROUTE_LOAD', 'DISABLED') === 'ENABLED';
+
+      if (autoLoadRoute) {
+        console.log('[MsfsFlightPlanSync] Configured to automatically load MSFS route - loading...');
+
+        this.handleSimRouteSent();
+      } else {
+        console.log('[MsfsFlightPlanSync] Configured to not automatically load MSFS route');
+      }
+
+      this.listener.on('AvionicsRouteSync', this.handleSimRouteSent);
+      this.listener.on('AvionicsRouteRequested', this.handleAvionicsRouteRequested);
+    });
+  }
+
+  private handleSimRouteSent = async (route?: JS_FlightPlanRoute): Promise<void> => {
+    route ??= await this.getSimRoute();
+
+    if (route.departureAirport.ident === '' || route.destinationAirport.ident === '') {
+      return;
+    }
+
+    const db = NavigationDatabaseService.activeDatabase;
+
+    await this.rpcClient.newCityPair(
+      route.departureAirport.ident,
+      route.destinationAirport.ident,
+      null,
+      FlightPlanIndex.Uplink,
+    );
+
+    if (route.departureRunway.number != '') {
```

Comment:
I think we should avoid using `!=` and use `!==` instead. Also check the other occurrences of `!=`.

Reply:
Oops.. Need a lint rule for this 😬